Replace hardcoded configurations by play vars

fwiesweg opened the pull request:
Hi,

I pulled some of the hard-coded constants into the playbook and command line arguments. If you feel like you'd like this upstream, let me know and I'll update the README. Fixes and improvements are of course welcome.

Best,
Florian

fwiesweg committed:
Replace hardcoded configurations by play vars

fwiesweg commented:
Oh and thanks for the good work! Really made our builds run quite a bit quicker.

spanezz commented:
Thanks, I like the direction this is going!

Unless I'm mistaken, nspawn-runner with this change would change behaviour on existing deployments. I wonder if the previous hardcoded values could be used as defaults when `nspawn_runner_*` keys are not present in the configuration, to allow smooth upgrades?

@kapouer, since you're also having nspawn-runner deployed, what do you think?

kapouer commented:
Indeed, keeping default behavior is good, unless there are better defaults which should be used.

fwiesweg committed:
Re-introduce previous default values

fwiesweg commented:
There you go ;-) I have not gotten around to test the latest commit with the old defaults in our installation, but the changes are quite straightforward.

fwiesweg committed:
Use systemd-nspawn binary to detect systemd version

Some systems (e.g. Arch) do not necessarily have the systemd binary on
PATH, so instead rely on something which we need to call later anyway.

fwiesweg committed:
Update README

fwiesweg commented:
Just tested it on our ci infrastructure and of course found some ... not so intelligent bugs. I fixed them all and updated the branch. README has been updated as well.

fwiesweg committed:
Pass systemd-nspawn arguments when running playbook
